### PR TITLE
Added disable user history/confirmation

### PR DESCRIPTION
**Pull request by asadazam93**
### Description
Added Disable user button with a confirmation pop up that allow you to add a comment/reason. Also added user enable/disable history

UI Images
Screenshot 2020-10-15 at 3 37 00 PM
Screenshot 2020-10-15 at 3 36 49 PM

**Sandbox**
N/A

### Reviewers
If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [ ] @awaisdar001 
- [ ] @DawoudSheraz 

### Post-review
- [ ] Rebase and squash commits

### CODE REVIEW

File: src/users/UserSummary.jsx [hunk 1]
```diff
@@ -14,11 +14,15 @@ export default function UserSummary({
 }) {
   const [ssoModalIsOpen, setSsoModalIsOpen] = useState(false);
   const [idvModalIsOpen, setIdvModalIsOpen] = useState(false);
+  const [disableUserModalIsOpen, setDisableUserModalIsOpen] = useState(false);
+  const [disableHistoryModalIsOpen, setDisableHistoryModalIsOpen] = useState(false);
```
Comment (DawoudSheraz) [inline src/users/UserSummary.jsx:18]: although the previous code is written is adding IsOpen at the end, in my opinion, the var name should be like IsDisableUserModelOpen. This gives the intent of being a boolean right from the start.

File: src/users/api.js [hunk 2]
```diff
@@ -161,7 +161,7 @@ export async function getAllUserData(userIdentifier) {
     enrollments = await getEnrollments(user.username);
     verificationStatus = await getUserVerificationStatus(user.username);
     ssoRecords = await getSsoRecords(user.username);
-    user.passwordStatus = await getUserPasswordStatus(user.username);
+    user = await getUserPasswordStatus(user.username);
```
Comment (DawoudSheraz) [inline src/users/api.js]: userPasswordStatus is more logical.

File: src/users/UserSummary.jsx [hunk 3]
```diff
@@ -249,6 +286,42 @@ export default function UserSummary({
...
+          open={disableUserModalIsOpen}
+          buttons={[
+            <Button
+              variant="success"
+              className="btn-outline-primary"
+              onClick={togglePasswordStatus}
+            >
+              Confirm
+            </Button>,
+          ]}
```
Comment (DawoudSheraz) [inline src/users/UserSummary.jsx]: Why two modals?

**Review by DawoudSheraz — COMMENTED** [review]
Please add how the UI is looking with button and modal.

File: src/users/api.js (hunk 2 above)
Comment (asadazam93) [reply src/users/api.js]: I did not change the variable name. Previously it was only fetching the passwordStatus attribute from the api response but I needed all of the data

File: src/users/UserSummary.jsx (hunk 3 above)
Comment (asadazam93) [reply src/users/UserSummary.jsx]: Both modals are different.

File: src/users/UserSummary.jsx [hunk 4]
```diff
@@ -249,6 +291,42 @@ export default function UserSummary({
             />
           )}
         />
+        <Modal
+          open={disableHistoryModalIsOpen}
+          onClose={() => setDisableHistoryModalIsOpen(false)}
+          title="Enable/Disable History"
+          body={(
+            <Table
+              data={disableHistoryData}
```
Comment (awaisdar001) [inline src/users/UserSummary.jsx]: `defaultSortedColumn` and `defaultSortDirection`

File: src/users/UserSummary.jsx [hunk 5]
```diff
@@ -249,6 +291,42 @@ export default function UserSummary({
...
+              Confirm
+            </Button>,
+          ]}
+          onClose={() => setDisableUserModalIsOpen(false)}
+          title={`${userData.status === PASSWORD_STATUS.USABLE ? 'Disable user confirmation' : 'Enable user confirmation'}`}
+          body={(
+            <div>
+              <label htmlFor="userIdentifier">Reason: </label>
+              <Input
+                name="userIdentifier"
```
Comment (awaisdar001) [inline src/users/UserSummary.jsx]: userIdentifier`

File: src/users/UserSummary.jsx [hunk 6]
```diff
@@ -262,7 +340,8 @@ UserSummary.propTypes = {
     isActive: PropTypes.bool,
     country: PropTypes.string,
     dateJoined: PropTypes.string,
-    passwordStatus: PropTypes.string,
+    status: PropTypes.string,
+    disableHistory: PropTypes.shape([]),
```
Comment (awaisdar001) [inline src/users/UserSummary.jsx]: `disableHistory`

File: src/users/api.js (hunk 2 above)
Comment (awaisdar001) [reply src/users/api.js]: ++

File: src/users/UserSummary.jsx (hunk 4 above)
Comment (asadazam93) [reply src/users/UserSummary.jsx]: A user probably will not have a long history of disabling/enabling so I guess this would be an overhead

**Review by DawoudSheraz — APPROVED** [review]

**Review by awaisdar001 — APPROVED** [review]

**Pull request merged**